src/main/java/io/github/fusionflux/portalcubed/content/prop/ButtonActivatedProp.java

```diff
@@ -0,0 +1,56 @@
+package io.github.fusionflux.portalcubed.content.prop;
+
+import io.github.fusionflux.portalcubed.content.button.FloorButtonBlock;
+import net.minecraft.CrashReport;
+import net.minecraft.CrashReportCategory;
+import net.minecraft.ReportedException;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.level.Level;
+
+public class ButtonActivatedProp extends Prop {
+	public ButtonActivatedProp(PropType type, EntityType<?> entityType, Level level) {
+		super(type, entityType, level);
+	}
+
+	public void setActivated(boolean activated) {
+		setVariantFlag(0, activated);
+	}
+
+	@Override
+	protected int dirtyFlagIndex() {
+		return 1;
+	}
+
+	@SuppressWarnings("deprecation")
+	@Override
+	protected void checkInsideBlocks() {
```

Comment:
this logic should be done by the block with an optional interface on the entity

Reply:
Buttons can't track what entities are currently pressing them so if they activate cube they can't do activate them

Reply:
give them a power timer that counts down with each tick. Have the button keep resetting it

Reply:
That sounds worse than what is currently done, can you explain more why this is bad?

Reply:
so much code is duplicated, and it's hardcoded. Entities tick anyway so it has no impact on performance.